### ISSUE TITLE
Shutter crashes after taking a screenshot

### ISSUE THREAD

**Issue by aboledaink** [shutter-project/shutter#462]
#### Brief summary of issue
Shutter crashes after taking a screenshot, screenshot file is ok, but user interface 

#### Steps to reproduce the issue

1. open shutter
2. click on "Selection", select an area (I have a 3 second delay), let go of the left mouse button
3. press enter key to take screenshot
4. the program window dissapears (I am assuming it crashes)

If I run shutter from the terminal window, it seems to work better.
I do get an error

#### Error output
select was emitted by widget Gtk3::ToolButton=HASH(0x55c1b5a9cdd0)
Using monitor: 0 - 0 - 2560 - 1440

(shutter:10127): GdkPixbuf-CRITICAL **: 08:46:57.256: gdk_pixbuf_composite: assertion 'dest_x >= 0 && dest_x + dest_width <= dest->width' failed
Parsing wildcards for $name_%NNN
Parsed $width and $height: $name_%NNN
Parsed $profile: $name_%NNN
Parsed $name: Selection_%NNN
3 Ns used in wild-card
Searching for files with pattern: (?^u:Selection_(\d+)(?:\(\d+\))?\.png)
001 is currently greatest value...
002 is currently greatest value...
New uri after exists check: /home/agustin/capturas/Selection_003.png
Trying to save file to /home/agustin/capturas/Selection_003.png
Saving file /home/agustin/capturas/Selection_003.png, png, 9
Out of memory!
Segmentation fault (core dumped)

#### Extra information, such as Shutter version, display server in use (Xorg or Wayland), operating system and ideas for how to solve:
OS: Pop!_OS 20.04 LTS
Gnome: 3.36.8
display server: X11
![Selection_003](https://user-images.githubusercontent.com/76700351/161747482-cd7c6ee7-0933-476c-be39-716757b351e4.png)

**Comment by Photon89** [comment]
> Saving file /home/agustin/capturas/Selection_003.png, png, 9
> Out of memory!

Is it possible that the drive you are trying to save has no free space left?

**Comment by aboledaink** [comment]
hi, thank you for your response, but no.
![Selection_004](https://user-images.githubusercontent.com/76700351/161969739-9d2ab183-1797-41cc-a7b9-31288c6b91e1.png)

**Comment by aboledaink** [comment]
free ram is also not an issue
![Selection_005](https://user-images.githubusercontent.com/76700351/161969942-9c38b3e6-aad5-4380-98ca-ce1adf8a31aa.png)

**Comment by Photon89** [comment]
Disk space, not RAM! ;)

**Comment by aboledaink** [comment]
hi, I do have 654gb available, see previous screenshot :)
There is only a single 1TB nvme (gen3) ssd drive that is completly encrypted, I don't know if that helps?

**Comment by Photon89** [comment]
Sorry, I missed the first comment. /home is just a subfolder in /, not a partition, right? You also do have write permission in the folder?

**Comment by aboledaink** [comment]
yes, /home is just a subfolder and the folder where the screenshots are saved is /home/agustin/capturas
This is my home folder.

But screenshots are saved. It seems that the program crashes after that. When I restart shutter it does not show the tab with the new screenshot. Does that help?

Let me recap:
1. open shutter
2. click on "Selection", select an area (I have a 3 second delay), let go of the left mouse button
3. press enter key to take screenshot
4. screenshot file is saved in /home/agustin/capturas
5. the program window dissapears (I am assuming it crashes)
6. I restart shutter and it does not show the new screenshot (I can load it if I want to)

**Comment by Photon89** [comment]
Did you install Shutter via PPA or, possibly, some Snap, Flatpak or similar?

**Comment by aboledaink** [comment]
I am not sure? I can see it in synaptic
![image](https://user-images.githubusercontent.com/76700351/162023560-2d3740f2-31f8-454c-9867-d119f77f245c.png)
but I can't find it in the Pop!_shop

Ii seems  I added this ppa and installed from it?
![image](https://user-images.githubusercontent.com/76700351/162024207-e517d062-17a4-444b-9bd8-ccb432c92f1e.png)

**Comment by Photon89** [comment]
Looks like it is installed via PPA, so that's fine... Does the problem occur every time you try to take a screenshot or only sporadically? If the latter is the case, does it rather occur with larger screenshot sizes or does it occur with small selections equally frequently? Did the problem appear from the very beginning or did it start appearing after some package update?

**Comment by aboledaink** [comment]
It seems to happen frequently, but not always. This morning I tried taking 3 screenshots and it happened all 3 times. Then later I made a couple more and all went fine.
I can't say if it happens more often with small screenshots or large screenshots. I do not take very large screenshots often. My main use is to ilustrate emails :)

**Comment by DarthGandalf** [comment]
I recall encountering something like that, but didn't dig further. It's not
a crash though, because the process is still running, just the window
doesn't reappear.
If I not mistaken, the tray icon was also missing, but I m not sure anymore.

пн, 11 апр. 2022 г., 19:24 aboledaink ***@***.***>:

**Comment by Photon89** [comment]
I have some spare time and could add some debug output, so we at least know which code line throws this error. Unless this is clear already? :smile:

**Comment by aboledaink** [comment]
> I recall encountering something like that, but didn't dig further. It's not a crash though, because the process is still running, just the window doesn't reappear. If I not mistaken, the tray icon was also missing, but I m not sure anymore. пн, 11 апр. 2022 г., 19:24 aboledaink ***@***.***>:

the tray icon does disapear, but the shutter process is not shown anymore. So I think it does crash?

**Comment by scaleoutsean** [comment]
- v0.99.2 on Mint 20.3 (i.e. Ubuntu 20.04)

I don't know if it's the same problem or not, but in my case Shutter tray icon remains, but unresponsive (I have to kill it and restart Shutter). The way I repro is use Selection, click to select screen area that's very small and rapidly click to take a screenshot after that. I'm not sure what's the relevant part here, it may be that screenshot width isn't larger than 0?

System log:

```
Jul  5 17:04:19 user shutter[2654400]: gdk_pixbuf_new_from_data: assertion 'width > 0' failed
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Shutter::Screenshot::SelectorAdvanced::select_advanced(Shutter::Screenshot::SelectorAdvanced=HASH(0x5576da55c260)) called at /usr/bin/shutter line 5991
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Shutter::App::fct_take_screenshot(Gtk3::ImageMenuItem=HASH(0x5576d9aaeb20), "tray_select", undef, undef) called at /usr/bin/shutter line 3065
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5576d97e7ec8)) called at /usr/bin/shutter line 10943
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Shutter::Screenshot::SelectorAdvanced::select_advanced(Shutter::Screenshot::SelectorAdvanced=HASH(0x5576da55c260)) called at /usr/bin/shutter line 5991
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Shutter::App::fct_take_screenshot(Gtk3::ImageMenuItem=HASH(0x5576d9aaeb20), "tray_select", undef, undef) called at /usr/bin/shutter line 3065
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: #011Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5576d97e7ec8)) called at /usr/bin/shutter line 10943
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: ***   Can't call method "get_width" on an undefined value at /usr/bin/shutter line 7700.
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: ***   #011Shutter::App::fct_parse_filename_wildcards("\$name_%NNN", Shutter::Screenshot::SelectorAdvanced=HASH(0x5576da55c260), undef) called at /usr/bin/shutter line 6276
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: ***   #011Shutter::App::fct_take_screenshot(Gtk3::ImageMenuItem=HASH(0x5576d9aaeb20), "tray_select", undef, undef) called at /usr/bin/shutter line 3065
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]: ***   #011Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5576d97e7ec8)) called at /usr/bin/shutter line 10943
Jul  5 17:04:19 user org.mate.panel.applet.MintMenuAppletFactory[2654400]:  at /usr/bin/shutter line 10943.
```

**Comment by Photon89** [comment]
It's a different issue indeed, at least the error output is completely different, so I created a new ticket to track it.
